packages/ember-application/lib/system/application.js

```diff
@@ -1003,7 +1003,7 @@ Application.reopenClass({
     registry.optionsForType('component', { singleton: false });
     registry.optionsForType('view', { singleton: false });
     registry.optionsForType('template', { instantiate: false });
-    registry.optionsForType('helper', { instantiate: false });
+    registry.optionsForType('helper', { singleton: false });
```

Comment:
I think the singleton part isn't that important. The rest of the system should prefer `lookupFactory(helperName).create(..)` ?